presto-main/src/main/java/com/facebook/presto/tracing/SimpleTracerBlock.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tracing;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+public class SimpleTracerBlock
+{
+    private final List<SimpleTracerPoint> points = new CopyOnWriteArrayList<>();
+    private final long startTime = System.currentTimeMillis();
```

Comment:
Note: `System.currentTimeMillis()` is the current wall clock time, and importantly for the purposes of tracing: is not monotonic. That means that things like an NTP update or daylight savings time on the host clock can skew measured time. Prefer `System.nanoTime()` instead which is monotonic (but might return negative values, so `System.nanoTime() - startTime` is the preferred way to determine elapsed time).

Reply:
@pettyjamesm, interesting insight. @tanjialiang, could you help to put up a PR to have a quick fix?